storage_service/locations/models/s3.py

```diff
@@ -128,12 +128,29 @@ def browse(self, path):
         }
 
     def delete_path(self, delete_path):
-        objects = self.resource.Bucket(self._bucket_name()).objects.filter(
+        """Delete an object from an S3 bucket. We assume an object exists, if
+        it doesn't then the generator returned by the S3 library (Boto3) cannot
+        be iterated, and we raise a StorageException.
+        """
+        if delete_path.startswith(os.sep):
+            LOGGER.info(
+                "S3 path to delete {} begins with {}; removing from path prior to deletion".format(
+                    delete_path, os.sep
+                )
+            )
+            delete_path = delete_path.lstrip(os.sep, "")
+        obj = self.resource.Bucket(self._bucket_name()).objects.filter(
             Prefix=delete_path
         )
-
-        for objectSummary in objects:
-            objectSummary.delete()
+        items = False
+        for objectSummary in obj:
+            items = True
+            resp = objectSummary.delete()
+            LOGGER.info("S3 response when attempting to delete: %s", resp)
```

Comment:
Maybe this is more debug level logging? Here's what I get when testing with minio for a success response:

```INFO      2019-06-13 12:35:13  locations.models.s3:s3:delete_path:149:  S3 response when attempting to delete: {'ResponseMetadata': {'HTTPStatusCode': 204, 'RetryAttempts': 0, 'HostId': '', 'RequestId': '15A7D91CB99426C4', 'HTTPHeaders': {'x-xss-protection': '1; mode=block', 'x-minio-deployment-id': 'b7808373-b4c1-41bb-929d-bed6c135c7e6', 'content-security-policy': 'block-all-mixed-content', 'accept-ranges': 'bytes', 'vary': 'Origin', 'server': 'MinIO/RELEASE.2019-05-14T23-57-45Z', 'date': 'Thu, 13 Jun 2019 19:35:13 GMT', 'x-amz-request-id': '15A7D91CB99426C4', 'x-amz-bucket-region': 'minio'}}}```

I can't find the documentation (boto3's is pretty useless), but I believe that if there's an error deleting the object (e.g. permisisons) then object.delete should raise an exception.

Reply:
Good idea!

Reply:
RE: the exception. In the specific case we're writing this code against, Boto3 isn't throwing an exception. This might be the version of the library at time of writing which is why we're doing this in a more ask-permission kind of way. 
```
   11: boto3==1.6.5
   12: botocore==1.9.23          # via boto3, s3transfer
```
We have another PR in the works where Boto3 is upgraded. Once that is done and we reach code-freeze how we handle this exception will come out in testing.